case.go

```diff
@@ -0,0 +1,39 @@
+package main
+
+type TestCase struct {
+	Client string
+	Params map[string]string
+}
+
+func Collect(matrix Matrix) []TestCase {
+
+	cases := make([]TestCase, 0)
```

Comment:
`nil` is a valid slice. use:

    var cases []TestCase

https://github.com/golang/go/wiki/CodeReviewComments#declaring-empty-slices